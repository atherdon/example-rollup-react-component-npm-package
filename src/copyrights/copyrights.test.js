import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Copyright from './index';

configure({ adapter: new Adapter() });


describe('copyrights/index component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Copyright />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly (Snapshot)', () => {
        const wrapper = renderer.create(<Copyright />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('<em> has correct data attribute', () => {
        const wrapper = shallow(<Copyright />);
        const em = wrapper.find('em');
        expect(em.props()).toHaveProperty('data-testid', 'copyTest');
    });

    it('Checking <em> text', () => {
        const wrapper = shallow(<Copyright />);
        const em = wrapper.find('em');
        expect(em.text()).toEqual(`Copyright Â©${new Date().getFullYear()} Hacker Noon. All rights reserved.`)
    })
});
