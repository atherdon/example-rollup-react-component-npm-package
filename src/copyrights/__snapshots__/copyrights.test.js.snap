// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`copyrights/index component renders correctly (Snapshot) 1`] = `
<em
  data-testid="copyTest"
>
  Copyright Â©
  2021
   
  Hacker Noon. All rights reserved.
</em>
`;
